learning_resources_search/serializers.py

```diff
@@ -130,6 +132,20 @@ def serialize_learning_resource_for_update(
             SearchCourseNumberSerializer(instance=num).data
             for num in learning_resource_obj.course.course_numbers
         ]
+
+    if learning_resource_obj.in_featured_lists > 0:
+        featured_rank = (
+            LearningResourceRelationship.objects.filter(
+                child_id=learning_resource_obj.id, parent__channel__isnull=False
+            )
+            .order_by("position")
+            .first()
+            .position
```

Comment:
Can `featured_rank` also be calculated via an `annotate` call in the `ViewSet` query `for_search_serialization` function instead?  re https://github.com/mitodl/mit-open/pull/1303:

> We should not be performing database queries in serializers going forward (aside from validation, create(), and update()).

Reply:
I can play around with `Case` and `Subquery` to try to make it work but I don't think it would just make the code more complicated without any benefits to performance. Even though the function is called `serialize_learning_resource_for_update` , it's not really a serializer in the sense that it returns data to a user that is waiting for it, it's just used to convert a learning resources into the format expected by open search.  featured_rank is only calculated for the double digit number of resources that are actually in a featured list so this does also really slow down the recreate index job